### DISCUSSION TITLE
Add default type function to metadata dictionary

### DISCUSSION THREAD

**Discussion by 28106830** [godotengine/godot-proposals#7317] [Editor]
### Describe the project you are working on

Metadata Add Dictionary

Can we make it the first line and add the type? After adding the type, the default values added later are all of this type. If not manually modified

### Describe the problem or limitation you are having in your project

Adding metadata to the dictionary is inconvenient to write

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add default type function to metadata dictionary

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Null

### If this enhancement will not be used often, can it be worked around with a few lines of script?

This feature is frequently used

### Is there a reason why this should be core and not an add-on in the asset library?

Yes

**Comment by AThousandShips** [comment]
All the questions are obligatory, if you are unable to fill them in open a [discussion](https://github.com/godotengine/godot-proposals/discussions) instead

I will convert this and your other proposal accordingly

**Comment by Calinou** [comment]
I've just tried creating metadata in 4.1.1.stable and can notice the type field is preserved across additions (the Rect2 type was already selected here):

![image](https://github.com/godotengine/godot-proposals/assets/180032/d901fc02-6d4e-4348-b8ab-46996d480740)

This isn't preserved across editor restarts though.
